// react-oauth2-login/lib/react-oauth2-login.jsx
import React, { PropTypes, Component } from 'react';
import { Provider, Request } from 'oauth2-client-js';
import './react-oauth2-login.scss';

class OauthLogin extends Component {
  login = () => {

    const { providerName, clientId, authorizationUrl, redirectUri, scope } = this.props;

    const provider = new Provider({
      id: providerName,
      authorization_url: authorizationUrl,
    });
    const request = new Request({
      client_id: clientId,
      redirect_uri: redirectUri,
      scope: scope,
    });

    const uri = provider.requestToken(request);
    provider.remember(request);
    window.location.href = uri;
  }

  render() {
    const {tag, buttonText} = this.props;
    const OauthLoginButton = React.createElement(
      tag, {
        onClick: this.login,
        className: 'react-oauth2-login'
      }, buttonText
    );
    return OauthLoginButton;
  }
}

OauthLogin.propTypes = {
  buttonText: PropTypes.string,
  tag: PropTypes.string,
  providerName: PropTypes.string,
  clientId: PropTypes.string,
  authorizationUrl: PropTypes.string ,
  redirectUri: PropTypes.string,
  scope: PropTypes.string,
};

OauthLogin.defaultProps = {
  tag: 'button',
  buttonText: 'Login with Oauth'
};

export default OauthLogin;
